src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarLoreUtil.java

```diff
@@ -0,0 +1,399 @@
+                colorNeutral(Translation.of("siege_lore_shield_name")));
+
+        meta.setDisplayName(name);
+
+        setBannerLore(meta, holder.getPersistentDataContainer());
+
+        stack.setItemMeta(meta);
+    }
+
+    public static void setBannerStackFromHolder(ItemStack stack, PersistentDataHolder holder) {
+        if (stack == null || holder == null) return;
+        if (!stack.hasItemMeta()) return;
+
+        ItemMeta meta = stack.getItemMeta();
+
+        copyData(holder, meta);
+
+        meta.getPersistentDataContainer().set(LORE_ITEM, PersistentDataType.STRING, SIEGE_BANNER);
+
+        String name = Translation.of("siege_lore_format_two_values",
+                getFormattedTown(holder.getPersistentDataContainer()),
+                colorNeutral(Translation.of("siege_lore_banner_name")));
+
+        meta.setDisplayName(name);
+
+        setBannerLore(meta, holder.getPersistentDataContainer());
+
+        stack.setItemMeta(meta);
+    }
+
+    private static void setBannerLore(ItemMeta meta, PersistentDataContainer data) {
+        List<String> lore = new ArrayList<>();
+
+        lore.add(getFormattedType(data));
+
+        String attacker_line = Translation.of("siege_lore_format_key_value",
+                colorKey(Translation.of("siege_lore_attacker")),
+                Translation.of("siege_lore_format_two_values",
+                        getFormattedAttacker(data),
+                        colorValue(Translation.of("siege_lore_format_secondary_value", getFormattedAttackerPoints(data)))));
+        lore.add(attacker_line);
+
+        String defender_line = Translation.of("siege_lore_format_key_value",
+                colorKey(Translation.of("siege_lore_defender")),
+                Translation.of("siege_lore_format_two_values",
+                        getFormattedDefender(data),
+                        colorValue(Translation.of("siege_lore_format_secondary_value", getFormattedDefenderPoints(data)))));
+        lore.add(defender_line);
+
+        String winner_line = Translation.of("siege_lore_format_key_value",
+                colorKey(Translation.of("siege_lore_winner")),
+                Translation.of("siege_lore_format_two_values",
+                        getFormattedWinner(data),
+                        colorValue(Translation.of("siege_lore_format_secondary_value", getFormattedStatus(data)))));
+        lore.add(winner_line);
+
+        String start_line = Translation.of("siege_lore_format_key_value",
+                colorKey(Translation.of("siege_lore_start")),
+                colorValue(getFormattedStart(data)));
+        lore.add(start_line);
+
+        String end_line = Translation.of("siege_lore_format_key_value",
+                colorKey(Translation.of("siege_lore_end")),
+                colorValue(getFormattedEnd(data)));
+        lore.add(end_line);
+
+
+        meta.setLore(lore);
+    }
+
+    public static void copyData(PersistentDataHolder fromHolder, PersistentDataHolder toHolder) {
```

Comment:
Needs a more descriptive method name, what data is copied?

Reply:
Renamed to show it copies PersistentData to a PersistentDataHolder, unsure on the exact name so not marked resolved